source/grammar/index.rst

```diff
@@ -1,12 +1,132 @@
+    classical_type_and_id_list: ( declare_type association "," )* declare_type association
+    declare_type: dependent_type_declaration 
+        :| independent_type_specifier
+    dependent_type_declaration: dependent_type_specifier designator
+    dependent_type_specifier: "int" | "uint" | "angle" | "fixed" | "stretch"
+    designator: "[" expression "]"
+    independent_type_specifier: "bit" | "creg" | "bool" 
+        :| timing_type
+    association: ":" id
+    return_signature: "->" classical_declaration
+    constant_declaration: "const" ( dependent_type_declaration | independent_type ) assignment 
+    classical_declaration: ( dependent_type_declaration | independent_type ) assignment?
+    program_block: "{" ( program_block | statement ) "}"
+    kernel_declaration: "kernel" id classical_type_and_id_list return_signature? ";"
+    expression_statement: expression ";"
+        :| "return" expression ";
+    expression: primary_expression 
+        :| expression binaryop expression
+        :| unary_operator expression 
+        :| membership_test
+        :| expression '[' expression ']'
+        :| call "(" expression_list? ")"
+        :| expression incrementor
+        :| quantum_measurement
+        :| expression_terminator
+    expression_terminator: real | nninteger | id
+        :| constant
+        :| time_terminator
+    constant: "pi" | "π" | "tau" | "𝜏" | "euler"
+    expression_list: ( expression "," )* expression 
+    unary_operator: "~" | "!"
+    call: expression
+        :| builtin_math
+        :| cast_operator
+    builtin_math: "sin" | "cos" | "tan" | "exp" | "ln" | "sqrt" | "popcount" | "lengthof"
+    cast_operator: declare_type
+    incrementor: "++" | "--
+    pragma: "#pragma {" ... "}"
+    declaration_statement: (quantum_declaration | classical_declaration | constant_declaration) ";"
+    assignment_expression: expression assignment_operator expression
+    binary_operator: "+" | "-" | "*" | "/" 
+        :| "<<" | ">>" | "rotr" | "rotl" | "&" | "|" | "^" | "&&" | "||"
+        :| ">" | "<" | ">=" | "<=" | "==" | "!="
+    assignment_operator: "=" | "+=" | "-=" | "*=" | "/=" |
+        :| "<<=" | ">>=" |  "&=" | "|=" | "^=" | "~=" | "->"
+    membership_test: id "in" set_declaration
+    index_set_declaration: "{" index_set_generators "}"
+    index_set_generators: expression_list | range
+    range: expression? ":" expression? ( ":" expression )?
+    selection_statement: "if (" expression ")" program_block ( "else" program_block )?
+    iteration_statement: "for" membership_test program_block | "while (" expression ")" program_block
+    selection_directive_statement: selection_directive ";"
+    selection_directive: "break" | "continue" | "exit"
+    quantum_gate_definition: "gate" quantum_gate_signature quantum_gate_block
+    quantum_gate_signature: id quantum_gate_args id_lis
+    quantum_gate_args: ( "(" classical_type_and_id_list? ")" )?
```

Comment:
The spec says these are only angular--but maybe it's worth allowing general classical types?

Reply:
There is a duplicate `quantum_gate_args` below as well.

Reply:
> The spec says these are only angular--but maybe it's worth allowing general classical types?

Where does it say this?

Reply:
Ok sounds, good.